### ISSUE TITLE
ValueError after second post

### ISSUE THREAD

**Issue by dwissmann** [adamghill/django-unicorn#669]
Hi,
I am writing a simple app that creates tree structures (parent-child relationships) and display the parent and it's immediate children using a unicorn template and view - the underlying table is simple but it's also included below. The idea is that users can click on a child which loads the child as the parent with its children or click on the parent and load its parent (basic walking up and down a tree). The user can also add new children at any given time.

My issue: The app loads fine and the first click to one of the children properly reloads the view with the child being the parent. However any subsequent call fails with this error:
`ValueError: Cannot assign "1": "Tree.parent" must be a "Tree" instance.`

the model:

```
class Tree(models.Model):
    BRANCH_TYPE = {
        "B": "Branch",
        "L": "Leaf",
        "E": "External Choice",
    }
    title = models.CharField(max_length=255)
    desc = models.CharField(max_length=1024)
    type = models.CharField(max_length=1, choices=BRANCH_TYPE)
    external = models.CharField(max_length=1024)
    parent = models.ForeignKey('self', models.SET_NULL, null=True, blank=True, parent_link='children')
    
    def __str__(self):
        return self.title
```

the template:
```
<div>
    <div class="prompt" unicorn:click="walk_back()">{{ prompt.title }}</div>
    <hr>
    <ul>
    {% for option in child_options %}
    
        <li class="options" unicorn:click="walk({{ option.id }})">
            {{ option.title}}
        </li>
    {% endfor %}
        <li><input type="text" unicorn:model="title"><button unicorn:click="add_branch">Add</button></li>
    </ul>
</div>
```

the component:
```
from django_unicorn.components import UnicornView, QuerySetType
from treeMaker.models import Tree

class TreeWalkQView(UnicornView):
    option_id: int 
    prompt: Tree  
    title: str =''
    child_options: QuerySetType[Tree] = None
    
    def mount(self):
        self.option_id = 1
        self.prompt = Tree.objects.get(id=self.option_id)
        self.child_options = self.prompt.tree_set.all()
        
    def add_branch(self):
        self.prompt.tree_set.create(title=self.title, desc="new description", type="B")
        self.child_options = self.prompt.tree_set.all()
        self.title = ''
        
        
    def walk(self, new_option_id):
        self.option_id = new_option_id
        self.prompt = Tree.objects.get(id=self.option_id)
        self.child_options = self.prompt.tree_set.all()        
    
    def walk_back(self):
        self.prompt = self.prompt.parent
        self.option_id = self.prompt.id
        self.child_options = self.prompt.tree_set.all()  
```

It is so simple and yet it fails only after the first successful click - is this a bug or am I doing something fundementally wrong?

Best,

Dom W

**Comment by dwissmann** [comment]
unfortunately i am still stuck - there is definitely something strange going on. I changed the above code to include a child component (and view) and have the parent component loop through the "child records" calling the child component for each record. like that:
```
{% load unicorn %}
<div>
    <div class="prompt" unicorn:click="walk_back()">{{ prompt_title }}</div>
    <hr>
    <ul>
    {% for option in child_options %}
        {% unicorn 'tree-walk-options' option=option key=option.id %}
    {% endfor %}
        <li><input type="text" unicorn:model="title"><button unicorn:click="add_branch">Add</button></li>
    </ul>
</div>
```
with this child component:
```
from django_unicorn.components import UnicornView
import logging
log = logging.getLogger('django')


class TreeWalkOptionsView(UnicornView):
    option = None
    is_editing = False
    
    def walk(self, new_option_id):
        log.info("child option called")
        self.parent.walk(new_option_id)
        self.parent.force_render = True
        
    def edit(self):
        print(self.request.body)
        print(self.option)
        self.is_editing = True


    def cancel(self):
        self.is_editing = False

    def delete(self):
        self.option.delete()
        self.parent.force_render = True
    
    def save(self):
        self.option.save()
        self.is_editing = False
```

and the child view:
```
<li class="options">
    {% if is_editing %}
        <input type="text" unicorn:model.defer="option.title" />
        <button unicorn:click="save">Save</button>
        <button unicorn:click="cancel">Cancel</button>
        <button unicorn:click="delete">Delete</button>
    {% else %}
        <div  unicorn:click="walk({{ option.id }})">{{ option.title}}</div>
        <button unicorn:click="edit()">edit</button>
    {% endif %}
</li>
```

I can now successfully "walk the tree" (click on  a child to become the parent and back) but when I click on the edit button on the child view it doesn't change the DOM (even though I can tell from debugging that django renders and sends the new partials.

It seems to me that this and the example before is related - that django-unicorn "looses" it's context since the same page is reloaded with a different record.

Do I make sense?

Thanks

**Comment by dwissmann** [comment]
Disregard last comment - I think I got it work finally. I had to set the force render parent to true to "walk the tree" which made the walking successful however, when I wanted to just edit the child, the force parent variable was still set to true which forced the parent to "over write" the first call.
Suggestion: make sure the explicitly mention that such variables are preserved throughout the session.
